__init__.py

```diff
@@ -106,40 +113,93 @@ def respond(self, query):
         if len(query) == 0:
             return 0.0
 
-        r = ddg.query(query)
+        # note: '1+1' throws an exception
+        try:
+            r = ddg.query(query)
+        except:
+            LOG.warning("DDG exception!")
```

Comment:
Consider changing usages of `LOG` to `self.log`.

Reply:
Consider using `self.log.exception` instead of `self.log.warning` so that the stack trace is printed in the log.

Reply:
Converted LOGs to self.logs. I have considered using exception instead of warn, and the way I approach it is if it is a minor recoverable exception I log it using a warning. If it is an unrecoverable exception I use log.exception. I ask you to consider following this approach in the future as well lest users think the world is coming to an end even though nothing is really wrong.